networkx/algorithms/community/quality.py

```diff
@@ -330,3 +332,98 @@ def community_contribution(community):
         return L_c / m - out_degree_sum * in_degree_sum * norm
 
     return sum(map(community_contribution, communities))
+
+
+@require_partition
+def partition_quality(G, partition):
+    """Returns the coverage and performance of a partition of G.
+
+    The *coverage* of a partition is the ratio of the number of
+    intra-community edges to the total number of edges in the graph.
+
+    The *performance* of a partition is the ratio of the number of
+    intra-community edges plus inter-community non-edges with the total
+    number of potential edges.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+
+    partition : sequence
+        Partition of the nodes of `G`, represented as a sequence of
+        sets of nodes. Each block of the partition represents a
+        community.
+
+    Complexity
+    ----------
+    O(C^2 + L)
+        C = number of communities
+        L = links
+
+    Returns
+    -------
+    (float, float)
+        The (coverage, performance) tuple of the partition, as defined above.
+
+    Raises
+    ------
+    NetworkXError
+        If `partition` is not a valid partition of the nodes of `G`.
+
+    Notes
+    -----
+    If `G` is a multigraph;
+        - for coverage, the multiplicity of edges is counted
+        - for performance, the result is -1 (total number of possible edges is not defined)
+
+    References
+    ----------
+    .. [1] Santo Fortunato.
+           "Community Detection in Graphs".
+           *Physical Reports*, Volume 486, Issue 3--5 pp. 75--174
+           <https://arxiv.org/abs/0906.0612>
+    """
+
+    node_community = {}
+    for i, community in enumerate(partition):
+        for node in community:
+            node_community[node] = i
+
+    # `performance` is not defined for multigraphs
+    if not G.is_multigraph():
+        # Iterate over the communities, quadratic, to calculate `possible_inter_community_edges`
+        possible_inter_community_edges = 0
+        for i in range(0, len(partition) - 1):
+            for j in range(i + 1, len(partition)):
+                possible_inter_community_edges += len(partition[i]) * len(partition[j])
```

Comment:
Probably more performant to push the loops into a library like `itertools.combinations`
```suggestion
        possible_inter_community_edges = sum(len(p1) * len(p2) for p1, p2 in itertools.combinations(partitions, 2))
```